src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -90,27 +95,44 @@ public LanguageWorkerState CreateWorkerStateWithExistingChannel(string language,
             return state;
         }
 
-        public LanguageWorkerState CreateWorkerState(string language)
+        public async Task InitializeAsync(string workerRuntime, IEnumerable<FunctionMetadata> functions)
         {
-            ILanguageWorkerChannel initializedChannel = _languageWorkerChannelManager.GetChannel(language);
+            workerRuntime = workerRuntime ?? Utility.GetWorkerRuntime(functions);
+            if (Utility.ShutdownStandbyChannels(_environment, functions, workerRuntime))
+            {
+                await _languageWorkerChannelManager.ShutdownChannelsAsync();
```

Comment:
I'm confused that your "should" check above is taking in a workerRuntime, but the actual shutdown method isn't - it's recomputing the runtime from env.